### ISSUE TITLE
ViewComponentResult does not await the executor.ExecuteAsync method.

### ISSUE THREAD

**Issue by gdegeneve** [aspnet/Mvc#4998]
If you return a ViewComponentResult from a Controller, it behaves like a fire and forget. The request pipeline does not wait for the ViewComponent to finish rendering. This Bug is new with the 1.0.0 release. Depending on the timings, this results in a half rendered view component result on the client and exceptions from the MemoryPoolViewBufferScope.

A followup bug would be, that if you inject a scoped service within that abandoned ViewCompoent it passes you the service instance from another concurrently processed request.

I implemented a workaround in my project by returning a ViewResult, that processes a .cshtml file which in turn displays the ViewComponent via await Component.InvokeAsync(..).

**Comment by rynowak** [comment]
@gdegeneve do you have a repro project we can take a look at?

**Comment by gdegeneve** [comment]
I published a small example that illustrates this bug, if it is one. The index page of the sample also includes a description.
https://github.com/gdegeneve/ViewComponentBug

**Comment by rynowak** [comment]
@gdegeneve - I can confirm that this is a bug. Thanks for the really thorough report, this would have been very hard to track down otherwise.

You can work around the problem like so.
```
services.AddMvc();

// This will replace the default registration of ViewComponentResultExecutor
services.AddTransient<ViewComponentResultExecutor>();
```

**Comment by gdegeneve** [comment]
Thanks for the workaround. This is better than to use a dummy View in between the ViewComponent and the Controller ;)

**Comment by rynowak** [comment]
Fixed in 1.0.1: 85ca3e4976d841425412fb987a6f38dbbaa0a6eb
Fixed in 1.1.0: bc76c0ef316dfa08c613ba9516ceb04848092491
